Add timeout support and more baud rates

jacobsparts opened the pull request:
timeout support via uselect
inWaiting method doesn't work for me, otherwise that could be used instead of uselect

jacobsparts committed:
Add timeout support and more baud rates

timeout support via uselect

pfalcon commented on serial.py:
```diff
@@ -22,12 +23,17 @@ class SerialDisconnectException(SerialException):
 class Serial:
 
-    BAUD_MAP = { 9600: termios.B9600, 115200: termios.B115200 }
+    BAUD_MAP = {
```
Can you please split baud rate additions to a separate commit, as they're not related to the timeout changes?

jacobsparts replied:
done

pfalcon commented on serial.py:
```diff
@@ -22,12 +23,17 @@ class SerialDisconnectException(SerialException):
-    BAUD_MAP = { 9600: termios.B9600, 115200: termios.B115200 }
+    BAUD_MAP = {
+        9600: termios.B9600,
+        19200: 14,
```
Can you add comment what is 14/where it comes from?

jacobsparts replied:
termios baud rate constants have integer values which ultimately refer to positions in the tty baud rate table in the [linux kernel](https://github.com/torvalds/linux/blob/894025f24bd028942da3e602b87d9f7223109b14/drivers/tty/tty_baudrate.c).  Micropython's termios does not define all of the baud rate constants, but they can be referred to by their integer values.

pfalcon replied:
Thanks for the details, I actually added it as an inline comment too.

pfalcon commented on serial.py:
```diff
@@ -22,12 +23,17 @@ class SerialDisconnectException(SerialException):
-        assert timeout is None, "Only no-timeout mode is supported"
         self.port = port
         self.baudrate = baudrate
+        self.timeout = None if timeout is None else timeout * 1000
```
Did you actually test this? In MicroPython, poll() argument for "no timeout" is -1.

jacobsparts replied:
Yes, I did test it.  None and -1 produce the same behavior, but we can use -1 instead.

pfalcon reviewed (changes requested):
Looks good and useful! Few comments how to make it perfect. Let me know if you would go for the changes (if not, I may do that instead, as time permits).

jacobsparts committed:
add timeout support via uselect

jacobsparts committed:
add more baud rates

micropython's termios does not define the B19200 constant so we use the integer value instead.  The integer value refers to the position in the baud rate table found in the linux kernel's tty_baudrate.c

pfalcon commented:
Thanks for the refactors! I did few more cleanups and merged/released new version to PyPI.
